services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/helpers/Wrappers.groovy

```diff
@@ -0,0 +1,93 @@
+package org.openkilda.functionaltests.helpers
+
+import groovy.util.logging.Slf4j
+import org.junit.runners.model.MultipleFailureException
+
+@Slf4j
+class Wrappers {
+
+    /**
+     * Retries some operation for a given number of times. Retry is considered failed if any throwable is thrown.
+     * Throws MultipleFailureException if closure fails to execute in given amount of tries.
+     *
+     * @param times number of retries
+     * @param retryInterval pause between retries (in seconds)
+     * @param handler exception handler (receives exception on each failure)
+     * @param body operation to wrap around
+     * @return result of the body execution
+     */
+    def static retry(int times = 5, double retryInterval = 2, Closure handler = { e -> log.debug(e) }, Closure body) {
+        int retries = 0
+
+        List<Throwable> ex = []
+        while (retries++ < times) {
+            try {
+                return body.call()
+            } catch (Throwable t) {
+                ex.add(t)
+                handler.call(t)
+                sleep(retryInterval * 1000 as long)
```

Comment:
Would it be helpful to use` TimeUnit.SECONDS.sleep(2) ` here?

Reply:
This won't accept fractional number (say the default 0.5 value will become invalid)

Reply:
There are also MILLISECONDS, MICROSECONDS, NANOSECONDS. Maybe something from it could be helpful?

Reply:
In fact, it doesn't matter how to "sleep", it just improves readability.

Reply:
If I change it to MILLISECONDS, then
1. The unit will differ from 'timeout' unit (which is Seconds) and it will be better to change it too
2. The caller will have to specify both values in milliseconds, which is more verbose, less intuitive and at some point less readable (I prefer to think of timeouts/intervals in seconds)

Reply:
Np, Roman. I just give suggestions which could be helpful. If n * 1000 is more readable - I'm ok)